test/Spec.hs

```diff
@@ -168,18 +128,19 @@ parseValidate file java program = do
     return (d', rs_code, toml_code)
 
 -- compile a program that is supposed to fail compilation
-compileFailingProgram :: String -> String -> IO String
-compileFailingProgram file program =
-    (show <$> parseValidate file False program) `catch`
-             (\e -> return $ show (e::SomeException))
+compileFailingProgram :: String -> String -> IO()
+compileFailingProgram file program = do
+   prog <- (show <$> parseValidate file False program) `catch`
+               (\e -> let _ = (e :: SomeException) in return "")
+   assertBool "Compilation should have failed" ("" == prog)
```

Comment:
Are we going too far here? For programs that fail compilation ".ast" files actually contain error messages, which we want to check.